bgpd/bgp_mplsvpn.c

```diff
@@ -1421,6 +1421,18 @@ void vrf_unimport_from_vrf(struct bgp *bgp, struct bgp *vrf_bgp,
 		vpn_leak_postchange(idir, afi, bgp_get_default(), bgp);
 	}
 
+	/*
+	 * What?
+	 * So SA is assuming that since the ALL_LIST_ELEMENTS_RO
+	 * below is checking for NULL that export_vrf can be
+	 * NULL, consequently it is complaining( like a cabbage )
+	 * that we could dereference and crash in the listcount(..)
+	 * check below.
+	 * So make it happy, under protest, with liberty and justice
+	 * for all.
+	 */
+	assert(vrf_bgp->vpn_policy[afi].export_vrf);
```

Comment:
why not just test and return if null?

Reply:
I think the assert is fine, it makes it explicit that `vrf_bgp->vpn_policy[afi].export_vrf` shouldn't be NULL. A NULL check could be confusing here IMO.

Reply:
Agreed, an assert is the correct thing to do here if `export_vrf` should never be null.

Reply:
The current assert is really hard to understand.  Looking at https://ci1.netdef.org/browse/FRR-FRRPULLREQ-3286/artifact/shared/static_analysis/report-ab1e1f.html#EndPath
Suggest the assert be added before 1433:
    /* export_vrf should always be set when import_vrf is set */
    assert(vrf_bgp->vpn_policy[afi].export_vrf != NULL && vname != NULL);

This doesn't require a lot of reading to understand...

At *least* the comment should be added.
listnode_delete(vrf_bgp->vpn_policy[afi].export_vrf, vname);
--
1434 | XFREE(MTYPE_TMP, vname);

Reply:
I disagree, I explicitly put the assert before the for(ALL_...) because it was the bit of code causing the SA warning to happen.  And I put the big warning there so that if someone ever rewrote the code it would be in a spot, with a hopefully a big enough warning that they could make the decision about whether or not it can be removed in an intelligent fashion.

Reply:
change addresses my concern. thanks.  (FWIW asserts to keep AS happy don't make sense to me, asserts because of fubar code logic is reasonable -- beware of the Chaos Monkey.)

Reply:
They don't make a bunch of sense to me either, but we want to get the SA warnings in clang down to 0 so that we can have the CI system let us know when it finds something that is a problem.

Reply:
I'd be willing to discuss alternative ways to get a SA system that we can trust.